php/Block.php

```diff
@@ -63,66 +63,213 @@ public function register_block() {
 	 * @return string The markup of the block.
 	 */
 	public function render_callback( $attributes, $content, $block ) {
-		$post_types = get_post_types(  [ 'public' => true ] );
-		$class_name = $attributes['className'];
-		ob_start();
 
-		?>
-        <div class="<?php echo $class_name; ?>">
-			<h2>Post Counts</h2>
-			<ul>
-			<?php
-			foreach ( $post_types as $post_type_slug ) :
-                $post_type_object = get_post_type_object( $post_type_slug  );
-                $post_count = count(
-                    get_posts(
+		// Store this in a local var for caching since we'll reference it a few
+		// times in this method.
+		$current_post_id = get_the_ID();
+
+		/**
+		 * Fetch all public post types as objects, and construct a new array that
+		 * maps the post type's label name (not slug), and the number of
+		 * published posts within that post type so we can easily display it
+		 * below.
+		 */
+
+		// Array to store our new values in.
+		$public_post_type_counts = [];
+
+		// Get all public post type objects.
+		$post_type_objects = get_post_types(  [ 'public' => true ], 'objects' );
```

Comment:
I'm not sure `"public" => true` is right, but I don't have enough context on the business requirements to say otherwise.